core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -36,12 +36,26 @@ final class Pull[+F[_], +O, +R] private[fs2] (private val free: FreeC[F, O, R])
     new Pull(free.map(r => Right(r)).handleErrorWith(t => Result.Pure(Left(t))))
 
   /**
-    * Interpret this `Pull` to produce a `Stream`.
+    * Interpret this `Pull` to produce a `Stream`, introducing a scope.
     *
     * May only be called on pulls which return a `Unit` result type. Use `p.void.stream` to explicitly
     * ignore the result type of the pull.
     */
   def stream(implicit ev: R <:< Unit): Stream[F, O] = {
+    val _ = ev
```

Comment:
I guess we do this because of a compiler warning? Could we shut this warning in a better way? e.g., in Scala 2.13.2 we can use the `@nowarn` annotation and probably the silencer plugin in previous versions.

Reply:
Yeah fatal-warnings and warn-unused. I’m wary of adding compiler plugins but if only for 2.12, that’s okay. Though we don’t have fatal-warnings enabled on 2.12 so probably not necessary. There’s a lot of these we can fix with the 2.13 warning override. Want to open an issue for that?

Reply:
Will do that :+1: